### PR TITLE
Add task

### PR DESCRIPTION
**Pull request by A11yship**

### CODE REVIEW

File: src/components/Form/Form.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export default function Form() {
+	const addTask = useStore(state => state.addTask);
+
+	function handleSubmit(event) {
+		event.preventDefault();
+		const name = event.target.elements.task.value.trim();
+		const duration = Number.parseInt(event.target.elements.duration.value, 10);
+		addTask(name, duration);
+		event.target.reset();
```
Comment (ahohnsen) [inline src/components/Form/Form.js:13]: Nice! You can also write it like this:

```suggestion
        const form = event.target
		const name = form.elements.task.value.trim();
		const duration = Number.parseInt(form.elements.duration.value, 10);
		addTask(name, duration);
		form.reset();
```

Maybe in this way it is a bit easier to read 🙂

File: src/components/Form/Form.spec.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import Form from './Form';
+
+describe('Form component', () => {
+	it('should be displayed', () => {
+		render(<Form />);
+		const form = screen.getByRole('form');
+		expect(form).toBeInTheDocument();
+	});
+	it('should be clickable', async () => {
```
Comment (ahohnsen) [inline src/components/Form/Form.spec.js:17]: I would rephrase this as your submit-button is clickable and not the whole form 🤔 
```suggestion
	it('should have a clickable submit-button', async () => {
```

File: src/hooks/useStore.js [hunk 3]
```diff
@@ -6,6 +6,9 @@ const useStore = create(set => ({
 		{id: nanoid(), name: 'Abwaschen', time: 15},
 		{id: nanoid(), name: 'Post sortieren', time: 2},
 	],
+	addTask(name, time) {
+		set(({tasks}) => ({tasks: [{id: nanoid(), name, time}, ...tasks]}));
+	},
 
 	//Beispielcode
```
Comment (ahohnsen) [inline src/hooks/useStore.js:13]: Please delete all unused code.

**Review by ahohnsen — CHANGES_REQUESTED** [review]
Hey Hannah,

Well done 💪🏼 Please have a look at my comments.

Additionally, please go over your project and delete all files/code from the template which you do not use (e.g. Button-component, useFetch etc.) 🧹 🙂

File: src/hooks/useStore.js (hunk 3 above)
Comment (A11yship) [reply src/hooks/useStore.js:13]: I can delete the unused code in this branch? I forgot it at the beginning and thought I need an extra user story for that.

**Review by ahohnsen — APPROVED** [review]
👍🏼

File: src/components/Form/Form.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import StyledForm from './StyledForm';
+
+export default function Form() {
+	const addTask = useStore(state => state.addTask);
+
+	function handleSubmit(event) {
+		event.preventDefault();
+		const form = event.target;
+		const name = form.elements.task.value.trim();
```
Comment (macfrei) [inline src/components/Form/Form.js:11]: Why do you name the variable `name` and not `task`? In the end it is a name for a `task` right? :)

File: src/components/Form/Form.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+		const duration = Number.parseInt(form.elements.duration.value, 10);
+		addTask(name, duration);
+		form.reset();
+	}
+
+	return (
+		<StyledForm onSubmit={handleSubmit} name="task">
```
Comment (macfrei) [inline src/components/Form/Form.js:18]: Do you need the `name` attribute on the form?

File: src/components/Form/Form.spec.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+		render(<Form />);
+		const form = screen.getByRole('form');
+		expect(form).toBeInTheDocument();
+	});
+	it('should have a clickable submit-button', async () => {
+		const handleClick = jest.fn();
+		render(<Form onSubmit={handleClick} />);
+		const button = screen.getByText(`Speichern`);
+		await userEvent.click(button);
+		expect(handleClick).toHaveBeenCalled;
```
Comment (macfrei) [inline src/components/Form/Form.spec.js:22]: This test should not work as your `Form` does not have a prop `onSubmit`.

File: src/components/Form/Form.spec.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+		const task = 'Post';
+		const duration = '5';
+		render(<Form onSubmit={onSubmit} />);
+		const inputTask = screen.getByLabelText('Aufgabe');
+		const inputDuration = screen.getByLabelText('Dauer in Minuten');
+		const button = screen.getByRole('button', {name: /speichern/i});
+		await userEvent.type(inputTask, task);
+		await userEvent.type(inputDuration, duration);
+		await userEvent.click(button);
+		expect(onSubmit).toBeCalled;
```
Comment (macfrei) [inline src/components/Form/Form.spec.js:35]: Same here. Your `Form` does not have a prop `onSubmit` and it should not be called, I guess. I think you will need to mock Zustand for the tests here.

**Review by macfrei — COMMENTED** [review]
Good job, added a few comments though. 👍

File: src/components/Form/Form.js (hunk 5 above)
Comment (A11yship) [reply src/components/Form/Form.js:18]: Yes, the form geht her role only when it have a name. I use getByRole in the first test.

File: src/components/Form/Form.spec.js [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+		const button = screen.getByRole('button', {name: /speichern/i});
+
+		await userEvent.type(inputTask, task);
+		await userEvent.type(inputDuration, duration.toString());
+		await userEvent.click(button);
+
+		expect(addTask).toBeCalledTimes(1);
+		expect(addTask).toBeCalledWith(task, duration);
+	});
+	it('should not akzept whitespaces for the task', async () => {
```
Comment (ahohnsen) [inline src/components/Form/Form.spec.js:54]: Little typo 🙂 
```suggestion
	it('should not accept whitespaces for the task', async () => {
```

File: src/components/Form/Form.spec.js [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+		await userEvent.type(inputTask, task);
+		await userEvent.type(inputDuration, duration.toString());
+		await userEvent.click(button);
+
+		const invalidInput = container.querySelector('input:invalid');
+		expect(invalidInput).toBeInTheDocument();
+
+		expect(addTask).not.toHaveBeenCalled();
+	});
+	it('should not akzept a string for the duration', async () => {
```
Comment (ahohnsen) [inline src/components/Form/Form.spec.js:76]:
```suggestion
	it('should not accept a string for the duration', async () => {
```

File: src/components/Form/Form.spec.js [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+		await userEvent.type(inputTask, task);
+		await userEvent.type(inputDuration, duration);
+		await userEvent.click(button);
+
+		const invalidInput = container.querySelector('input:invalid');
+		expect(invalidInput).toBeInTheDocument();
+
+		expect(addTask).not.toHaveBeenCalled();
+	});
+	it('should not akzept negative duration', async () => {
```
Comment (ahohnsen) [inline src/components/Form/Form.spec.js:98]:
```suggestion
	it('should not accept negative duration', async () => {
```

**Review by ahohnsen — APPROVED** [review]
Nice tests 😍 Just a little typo.
